mycpp/mark_sweep_heap.cc

```diff
@@ -57,12 +57,18 @@ int MarkSweepHeap::MaybeCollect() {
 }
 
 // Allocate and update stats
-void* MarkSweepHeap::Allocate(size_t num_bytes) {
+// TODO: Make this interface nicer.
+void* MarkSweepHeap::Allocate(size_t num_bytes, int* obj_id, bool* in_pool) {
   // log("Allocate %d", num_bytes);
+  if (num_bytes <= pool_.kMaxObjSize) {
```

Comment:
Can you add #ifdef POOL_ALLOC around this and any other relevant parts?

I'm going to optimize StackRoots, and then I want to do some measurements with this

Reply:
Yep sure, done